Sourcecode/private/mx/core/Enums.cpp

```diff
@@ -3830,6 +3830,42 @@ namespace mx
             return toStream( os, value );
         }
 
+        /// UpDownNone /////////////////////////////////////////////////////////////////////////////////
+
+        UpDownNone parseUpDownNone( const std::string& value )
+        {
+            const auto opt = tryParseUpDownNone( value );
+            return opt.value_or( UpDownNone::none );
+        }
+
+        std::optional<UpDownNone> tryParseUpDownNone( const std::string& value )
+        {
+            if( value == "up" ) { return UpDownNone::up; }
+            else if( value == "down" ) { return UpDownNone::down; }
+            return std::optional<UpDownNone>{};
+        }
```

Comment:
I think you can do away with the std:optional by changing these parsing functions to return UpDownNone::none in the default case. Or is there a place where we need to know the difference between optional::none and UpDownNone::none?

Reply:
Sorry about the junk in the diff. I'll make a new one without the line ends and the sengo work that ended up in this diff.

I was afraid to dive in deeper to the use of the optional as other code uses the UpDown. The appregiate code could be smarter. As you state, it sets the default to Up, which add the direction attribute even if there was none present in the parsed XML.